### PR TITLE
Add Use Case for Undo, Redo and Add student

### PR DESCRIPTION
**Pull request by angkoonhwee**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -308,7 +308,84 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  * 1c1. TAB shows an error message for invalid email format.
+  * 1c2. User inputs the student's email address with the correct format.
+
+    Steps 1c1-1c2 are repeated until the email address inputted is a valid format.
+
+    Use case resumes from step 2.
+
+* 1d. User requests to add student with optional fields.
+
+  * 1d1. User inputs the optional fields with the respective prefixes together with the compulsory fields.
```
Comment (Xiaoyunnn) [inline docs/DeveloperGuide.md:350]: Should we add error messages for invalid format of the optional fields inputted?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -308,7 +308,84 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+1. User makes a request that <u>modifies the storage. E.g. ladd, add, edit, delete,...</u>.
+2. TAB modifies the storage.
+3. User requests to undo the last modification.
+4. TAB undoes the last command.
+
+   Use case ends.
+
+**Extension**
+* 1a. User did not make any request or User requests for commands that does not modify storage.
```
Comment (Xiaoyunnn) [inline docs/DeveloperGuide.md:366]: step 1a means this step is after step 1, so I think this step is not valid?

**Review by Xiaoyunnn — COMMENTED** [review]
I forgot that Chester is overseeing DG section😂 but here are some possibles considerations

**Comment by Chesterwongz** [comment]
Why doesn't this PR have auto CI checks?

**Review by Chesterwongz — APPROVED** [review]
Ran tests manually, locally

**Pull request merged**